ktor-http/common/src/io/ktor/http/URLBuilder.kt

```diff
@@ -86,33 +86,41 @@ public class URLBuilder(
         this.parameters = UrlDecodedParametersBuilder(encodedParameters)
     }
 
-    /**
-     * Build a URL string
-     */
-    // note: 256 should fit 99.5% of all urls according to http://www.supermind.org/blog/740/average-length-of-a-url-part-2
-    public fun buildString(): String = appendTo(StringBuilder(256)).toString()
-
     /**
      * Build a [Url] instance (everything is copied to a new instance)
      */
-    public fun build(): Url = Url(
-        protocol = protocol,
-        host = host,
-        specifiedPort = port,
-        pathSegments = pathSegments,
-        parameters = parameters.build(),
-        fragment = fragment,
-        user = user,
-        password = password,
-        trailingQuery = trailingQuery,
-        urlString = buildString()
+    public fun build(
+        defaultProtocol: URLProtocol = UrlProtocol.Http,
```

Comment:
This looks very counterintuitive that the `build` function accepts parameters, especially the ones that should be default. What does the usage of it look like?

Reply:
My idea was to prevent hardcoding the values, but it looks strange, thats right. Honestly, I can't imagining a usage with given parameters. I will move them into the build function body.